libchisel/src/dropsection.rs

```diff
@@ -0,0 +1,116 @@
+use super::{ModuleError, ModuleTranslator};
+
+use parity_wasm::builder::*;
+use parity_wasm::elements::*;
+
+/// Enum on which ModuleTranslator is implemented.
+pub enum DropSection<'a> {
+    NamesSection,
+    /// Name of the custom section.
+    CustomSectionByName(&'a String),
+    /// Index of the custom section.
+    CustomSectionByIndex(usize),
+    /// Index of the unknown section.
+    UnknownSectionByIndex(usize),
+}
+
+fn names_section_index_for(module: &Module) -> Result<usize, String> {
+    for (index, section) in module.sections().iter().enumerate() {
+        if let Section::Name(_section) = section {
+            return Ok(index);
+        }
+    }
+    Err("Not found".to_string())
+}
+
+fn custom_section_index_for(module: &Module, name: &String) -> Result<usize, String> {
+    for (index, section) in module.sections().iter().enumerate() {
+        if let Section::Custom(_section) = section {
+            if _section.name() == name {
+                return Ok(index);
+            }
+        }
+    }
+    Err("Not found".to_string())
+}
+
+impl<'a> DropSection<'a> {
+    fn find_index(&self, module: &Module) -> Result<usize, String> {
+        Ok(match &self {
+            DropSection::NamesSection => names_section_index_for(module)?,
```

Comment:
This was named `_for` because it had a section enum as a parameter to be generic. If we keep these helpers, they need better naming.